pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp

```diff
@@ -805,14 +804,16 @@ class StateVectorCudaManaged
         // allocate an external buffer if needed
         PL_CUSPARSE_IS_SUCCESS(cusparseSpMV_bufferSize(
             handle, CUSPARSE_OPERATION_NON_TRANSPOSE, &alpha, mat, vecX, &beta,
-            vecY, data_type, CUSPARSE_MV_ALG_DEFAULT, &bufferSize))
+            vecY, data_type, CUSPARSE_SPMV_CSR_ALG1,
+            &bufferSize)) // Can also use CUSPARSE_MV_ALG_DEFAULT
 
-        PL_CUDA_IS_SUCCESS(cudaMalloc(&dBuffer, bufferSize))
+        DataBuffer<void, int> dBuffer{bufferSize, device_id, stream_id, true};
 
         // execute SpMV
         PL_CUSPARSE_IS_SUCCESS(cusparseSpMV(
             handle, CUSPARSE_OPERATION_NON_TRANSPOSE, &alpha, mat, vecX, &beta,
-            vecY, data_type, CUSPARSE_MV_ALG_DEFAULT, dBuffer))
+            vecY, data_type, CUSPARSE_SPMV_CSR_ALG1,
```

Comment:
What is the advantage of `CUSPARSE_SPMV_CSR_ALG1` over `CUSPARSE_SPMV_ALG_DEFAULT` ?

Reply:
Not sure the advantage of CUSPARSE_SPMV_CSR_ALG1 but it's the Default algorithm for CSR sparse matrix format. While, CUSPARSE_SPMV_ALG_DEFAULT is  the Default algorithm for any sparse matrix format.
--


<br class="Apple-interchange-newline">